site/blocks/StickyNav.tsx

```diff
@@ -0,0 +1,179 @@
+        this.setHeadingPositions()
+    }, 100)
+
+    handleScroll = debounce(() => {
+        this.setCurrentHeading()
+    }, 100)
+
+    setCurrentHeading() {
+        const { scrollY } = window
+        let currentHeadingIndex = 0
+        for (let i = 0; i < this.state.headingPositions.length; i++) {
+            const heading = this.state.headingPositions[i]
+            const target =
+                heading.position -
+                STICKY_NAV_HEIGHT -
+                HEADING_SCROLL_MARGIN -
+                // Additional clearance for height of actual element
+                32
+            if (scrollY <= target) {
+                break
+            } else {
+                currentHeadingIndex = i
+            }
+        }
+
+        const current = this.ulRef.current
+        if (current) {
+            const parent = current.parentElement
+            if (parent) {
+                const hasScrollbar = parent.scrollWidth < current.scrollWidth
+                if (hasScrollbar) {
+                    const listItem = current.children[
+                        currentHeadingIndex
+                    ] as HTMLElement
+                    const containerPadding = 32
+                    current.scrollTo({
+                        behavior: "smooth",
+                        left: listItem.offsetLeft - containerPadding,
+                    })
+                }
+            }
+        }
+
+        this.setState({
+            currentHeadingIndex,
+        })
+    }
+
+    setHeadingPositions() {
+        const headingPositions: any[] = []
+        this.links.forEach((link) => {
+            const element = document.querySelector<HTMLElement>(link.target)
+            if (element) {
+                const { y } = getTotalOffset(element)
+                headingPositions.push({ id: link.target, position: y })
+            }
+        })
+
+        this.setState(
+            {
+                headingPositions,
+            },
+            () => {
+                this.setCurrentHeading()
+            }
+        )
+    }
+
+    componentDidMount() {
+        // Web fonts change the position of elements
+        // so we set the active heading only once we know where everything is.
+        // The alternative is to call setHeadingPositions as soon as the component mounts,
+        // but this can make currentHeadingIndex jump around.
+        window.addEventListener("load", this.handleResize)
+        window.addEventListener("scroll", this.handleScroll)
+        window.addEventListener("resize", this.handleResize)
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("scroll", this.handleScroll)
+        window.removeEventListener("resize", this.handleResize)
+        window.removeEventListener("load", this.handleResize)
+    }
+
+    render() {
+        if (!this.links) return null
+        return (
+            <>
+                <style>
+                    {`
+                    [id] {
```

Comment:
huh, where's this `[id]` coming from?

Reply:
It's some CSS magic to compensate for the sticky nav whenever it's present. I added a comment.